[DRAFT] Temporal climate enhancements (#129)
ACCESS-Community-Hub/PyEarthTools: docs/newproject.md

# New Project Guide

## A Quick Hands-On Approach

> This guide is suitable for scientists or anyone else who wants to start trying things quickly to establish their first model and make a first attempt. More detail is provided below with more detail on the nuances and alternatives for each step.

Reviewer: ```suggestion
This guide is suitable for scientists or anyone else who wants to quickly establish their first model, hands-on. The following describes the nuances and alternatives for each step, in detail.
```

1. Use [https://pyearthtools.readthedocs.io/en/latest/notebooks/tutorial/FourCastMini_Demo.html](https://pyearthtools.readthedocs.io/en/latest/notebooks/tutorial/FourCastMini_Demo.html) as a template for what to do.
1. Determine the parameters you want to model, such as `temperature` or `wind`. When these become part of the neural network, they will be called *channels*.

> 2. Determine the data source they come from, such as ERA5 or another model or re-analysis source

Reviewer: ```suggestion
2. Determine the data source they come from, such as ERA5 or another model or re-analysis source.
```

> 3. Develop a `pipeline` which includes data normalisation

Reviewer: ```suggestion
3. Develop a `pipeline` which includes data normalisation.
```

4. Using a bundled model, configure that model to the size required. This may only required the adjustment of `img_size`, `in_channels` and `out_channels` to match the size of your data. The grid dimension must be a multiple of four for this model, so you may need to crop or regrid your data to match. In future, a standard approach without this limitation will be added. 

> 5. Run some number of training steps (using the `.fit` method) and visualise the outputs. Visualising predictions from the trained model every 3000 steps or so provides useful insight into the training process as well as helping see when the model might be fully trained. *There is no definite answer to how much training will be required. If your model isn't showing any progress at all after a couple of epochs, there may be a problem. Some models will start to show progress after 3000 steps.*

Reviewer: ```suggestion
5. Run some number of training steps (using the `.fit` method) and visualise the outputs. Visualising predictions from the trained model every 3000 steps or so is useful in providing insight into the training process, and for understanding when the model might be fully trained. *There is no definite answer to how much training will be required. If your model isn't showing any progress at all after a couple of epochs, there may be a problem. Some models will start to show progress after 3000 steps.*
```
